src/components/App/App.js

```diff
@@ -1,21 +1,52 @@
-import React from 'react'
+import React, {useReducer} from 'react'
+import fakeData from  '../../dummy_data.json'
 import { Routes, Route } from 'react-router-dom'
 import { UserDashboard } from '../User_Dashboard/User_Dashboard'
 import { Login } from '../Login/Login'
 import { SearchResults } from '../Search_Results/Search_Results'
 import { FriendsGames } from '../Friends_Games/Friends_Games'
 import { Navbar } from '../Navbar/Navbar'
 
+const initialState = {
+  searchResults : [],
+  user: {},
+  userName: '',
+  friendsList: [],
+  gameCollection: [],
+  modalOpen: false,
+  error: null,
+  loading:false
+}
+
+const reducer = (state, action) => {
```

Comment:
This is great! If you ever do Redux, you'll see lots of similarities between its reducers and userReducer.